File: windows-client/astral_client/protocol_manifest.py
"""Classifies every server-to-client WS frame as HANDLED or IGNORED for the desktop
client so a new, unclassified frame fails the build rather than silently dropping;
MainWindow._on_message routes HANDLED types.
"""

from __future__ import annotations

HANDLED = "handled"
IGNORED = "ignored"

CLIENT_LOCAL_ACTIONS: frozenset[str] = frozenset({"attach_existing", "computer_host_consent"})

CLASSIFICATION: dict[str, str] = {
    "rote_config": HANDLED,
    "chrome_menu": HANDLED,
    "user_preferences": HANDLED,
    "system_config": IGNORED,
    "agent_list": HANDLED,
    "agent_registered": HANDLED,
    "agent_host_inventory_reconciled": HANDLED,
    "agent_host_registered": HANDLED,
    "agent_host_registration_refused": HANDLED,
    "agent_bundle_deliver": HANDLED,
    "agent_tunnel": HANDLED,
    "agent_stop": HANDLED,
    "agent_offline": HANDLED,
    "computer_request": HANDLED,
    "computer_session": HANDLED,
    "computer_host": HANDLED,
    "auth_required": HANDLED,
    "ui_render": HANDLED,
    "ui_update": IGNORED,
    "ui_upsert": HANDLED,
    "ui_append": IGNORED,
    "ui_stream_data": HANDLED,
    "chrome_render": HANDLED,
    "chrome_surface": HANDLED,
    "chat_status": HANDLED,
    "chat_step": HANDLED,
    "chat_created": HANDLED,
    "chat_loaded": HANDLED,
    "chat_deleted": IGNORED,
    "history_list": HANDLED,
    "user_message_acked": HANDLED,
    "task_started": HANDLED,
    "task_completed": HANDLED,
    "tool_progress": HANDLED,
    "workspace_timeline_mode": HANDLED,
    "conversation_commit_ready": HANDLED,
    "conversation_snapshot": HANDLED,
    "operation_status": HANDLED,
    "agent_lifecycle": HANDLED,
    "composer_state": HANDLED,
    "voice_control_binding": HANDLED,
    "voice_session_state": HANDLED,
    "voice_turn_state": HANDLED,
    "voice_submission_rejected": HANDLED,
    "voice_transcript": HANDLED,
    "voice_announcement_media": HANDLED,
    "voice_local_announcement": HANDLED,
    "voice_local_final_rejected": HANDLED,
    "voice_local_session_ready": HANDLED,
    "voice_local_turn_bound": HANDLED,
    "heartbeat": IGNORED,
    "stream_subscribed": HANDLED,
    "stream_unsubscribed": HANDLED,
    "stream_list": IGNORED,
    "stream_data": HANDLED,
    "stream_error": HANDLED,
    "component_saved": HANDLED,
    "component_save_error": HANDLED,
    "saved_components_list": HANDLED,
    "component_deleted": HANDLED,
    "combine_status": HANDLED,
    "combine_error": HANDLED,
    "components_combined": HANDLED,
    "components_condensed": HANDLED,
    "agent_permissions": IGNORED,
    "agent_permissions_updated": IGNORED,
    "llm_config_ack": IGNORED,
    "llm_usage_report": IGNORED,
    "audit_append": IGNORED,
    "agent_creation_progress": IGNORED,
    "notification": HANDLED,
    "error": HANDLED,
}


def is_handled(frame_type: str) -> bool:
    return CLASSIFICATION.get(frame_type) == HANDLED


def is_classified(frame_type: str) -> bool:
    return frame_type in CLASSIFICATION

File: android-client/core/src/main/kotlin/com/personalailabs/astraldeep/core/protocol/ProtocolManifest.kt
// Classifies every server-to-client WS frame type from the committed ui_protocol.json manifest as HANDLED or
// IGNORED, so ProtocolManifestTest fails the build when a new frame type isn't deliberately triaged.

package com.personalailabs.astraldeep.core.protocol

object ProtocolManifest {
    const val HANDLED = "handled"
    const val IGNORED = "ignored"

    val classification: Map<String, String> =
        mapOf(
            "rote_config" to IGNORED,
            "chrome_menu" to HANDLED,
            "user_preferences" to HANDLED,
            "system_config" to IGNORED,
            "agent_list" to HANDLED,
            "agent_registered" to IGNORED,
            "agent_host_inventory_reconciled" to IGNORED,
            "agent_host_registered" to IGNORED,
            "agent_host_registration_refused" to IGNORED,
            "auth_required" to HANDLED,
            "ui_render" to HANDLED,
            "ui_update" to IGNORED,
            "ui_upsert" to HANDLED,
            "ui_append" to IGNORED,
            "ui_stream_data" to HANDLED,
            "chrome_render" to IGNORED,
            "chrome_surface" to HANDLED,
            "chat_status" to HANDLED,
            "chat_step" to HANDLED,
            "chat_created" to HANDLED,
            "chat_loaded" to HANDLED,
            "chat_deleted" to IGNORED,
            "history_list" to HANDLED,
            "user_message_acked" to HANDLED,
            "task_started" to HANDLED,
            "task_completed" to HANDLED,
            "tool_progress" to HANDLED,
            "workspace_timeline_mode" to HANDLED,
            "conversation_snapshot" to HANDLED,
            "conversation_commit_ready" to HANDLED,
            "operation_status" to HANDLED,
            "agent_lifecycle" to HANDLED,
            "composer_state" to HANDLED,
            "voice_control_binding" to HANDLED,
            "voice_session_state" to HANDLED,
            "voice_turn_state" to HANDLED,
            "voice_submission_rejected" to HANDLED,
            "voice_transcript" to HANDLED,
            "voice_announcement_media" to HANDLED,
            "voice_local_announcement" to HANDLED,
            "voice_local_final_rejected" to HANDLED,
            "voice_local_session_ready" to HANDLED,
            "voice_local_turn_bound" to HANDLED,
            "heartbeat" to IGNORED,
            "stream_subscribed" to HANDLED,
            "stream_unsubscribed" to HANDLED,
            "stream_list" to IGNORED,
            "stream_data" to HANDLED,
            "stream_error" to HANDLED,
            "component_saved" to HANDLED,
            "component_save_error" to HANDLED,
            "saved_components_list" to HANDLED,
            "component_deleted" to HANDLED,
            "combine_status" to HANDLED,
            "combine_error" to HANDLED,
            "components_combined" to HANDLED,
            "components_condensed" to HANDLED,
            "agent_permissions" to IGNORED,
            "agent_permissions_updated" to IGNORED,
            "llm_config_ack" to IGNORED,
            "llm_usage_report" to IGNORED,
            "audit_append" to IGNORED,
            "agent_creation_progress" to IGNORED,
            "agent_tunnel" to IGNORED,
            "agent_bundle_deliver" to IGNORED,
            "agent_stop" to IGNORED,
            "agent_offline" to IGNORED,
            "computer_request" to IGNORED,
            "computer_session" to HANDLED,
            "computer_host" to HANDLED,
            "notification" to HANDLED,
            "error" to HANDLED,
        )

    fun isHandled(frameType: String): Boolean = classification[frameType] == HANDLED

    fun isClassified(frameType: String): Boolean = frameType in classification
}
